docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install

```diff
@@ -25,308 +29,444 @@ function su_humsci_profile_final_task() {
+
+    // Make sure not to copy the id or uuid fields from the ECK.
+    $skip_fields = ['id', 'uuid'];
+    // Copy any fields that exist on both entity types.
+    foreach (array_keys($entity->getFieldDefinitions()) as $field_name) {
+      if ($new_entity->hasField($field_name) && !in_array($field_name, $skip_fields)) {
+        $new_entity->set($field_name, $entity->get($field_name)->getValue());
+      }
+    }
+    $new_entity->save();
+    $sandbox['convert'][$eck_type][$id] = $new_entity->id();
+    $entity->delete();
+  }
+  $sandbox['#finished'] = empty($sandbox['items']) ? 1 : ($sandbox['count'] - count($sandbox['items'])) / $sandbox['count'];
 
-/**
- * Uninstall deprecated webform_jqueryui_buttons module.
- */
-function su_humsci_profile_update_8205() {
-  \Drupal::service('module_installer')->uninstall(['webform_jqueryui_buttons']);
-}
+  // When the conversion is complete, update all the field data.
+  if (!empty($sandbox['items'])) {
+    return;
+  }
 
-/**
- * Create the default migration group.
- */
-function su_humsci_profile_update_9000() {
-  $storage = \Drupal::entityTypeManager()
-    ->getStorage('migration_group');
-
-  if (!$storage->load('default')) {
-    $storage->create([
-      'uuid' => 'd97cf71e-57ce-428b-8247-003ce7d5166d',
-      'id' => 'default',
-      'label' => 'Default',
-    ])->save();
+  /** @var \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager */
+  $entity_field_manager = \Drupal::service('entity_field.manager');
+  $field_map = $entity_field_manager->getFieldMapByFieldType('entity_reference');
+  $update_fields = [];
+
+  // Find all the fields that reference each of the ECK entity types.
+  foreach ($field_map as $entity_type => $fields) {
+    foreach (array_keys($fields) as $field_name) {
+      $field_storage = FieldStorageConfig::loadByName($entity_type, $field_name);
+      if (!$field_storage) {
+        continue;
+      }
+      $target_type = $field_storage->getSetting('target_type');
+      if (isset($eck_info[$target_type])) {
+        $update_fields[$entity_type][$field_name] = $target_type;
+      }
+    }
   }
 
-  $config = \Drupal::configFactory()->getEditable('config_ignore.settings');
-  $ignored = $config->get('ignored_config_entities');
-  $ignored[] = 'hs_courses_importer.importer_settings';
-  $config->set('ignored_config_entities', $ignored)->save();
+  // Update the database for the new hs entity ids.
+  foreach ($update_fields as $entity_type => $update_entity_fields) {
+    foreach ($update_entity_fields as $field_name => $eck_type) {
+      $convert_ids = $sandbox['convert'][$eck_type] ?? [];
+
+      $query = \Drupal::database()
+        ->select("{$entity_type}__$field_name", 'f')
+        ->fields('f', ['entity_id', 'delta', "{$field_name}_target_id"])
+        ->execute();
+
+      while ($entry = $query->fetchAssoc()) {
+        $new_target_id = $convert_ids[$entry["{$field_name}_target_id"]] ?? FALSE;
+        if (!$new_target_id > 0) {
```

Comment:
I'm having a hard time understanding this conditonal. Is it checking if `$new_target_id` is not TRUE _and_ if it's greater than 0. Also wouldn't we want the $new_target_id to be a value greater than 0?

Reply:
I can't remember what the use case here was.  There was a site I tested on that had some orphans or something. It should always be greater than 0, but there were some that acted weird. I don't recall which site either.